In [1]:
!python -m prodigy drop news
!python -m prodigy drop news_train
!python -m prodigy drop news_eval


✘ Can't find 'news' in database SQLite


✘ Can't find 'news_train' in database SQLite


✘ Can't find 'news_eval' in database SQLite



In [2]:
!python -m prodigy db-in news annotated_news_headlines.jsonl

✔ Created dataset 'news' in database SQLite
✔ Imported 373 annotations to 'news' (session 2023-05-02_11-35-52) in
database SQLite
Found and keeping existing "answer" in 373 examples


In [3]:
!python -m prodigy train --ner news

ℹ Using CPU

========================= Generating Prodigy config =========================
ℹ Auto-generating config with spaCy
✔ Generated training config

=========================== Initializing pipeline ===========================
[2023-05-02 11:35:53,300] [INFO] Set up nlp object from config
Components: ner
Merging training and evaluation data for 1 components
  - [ner] Training: 298 | Evaluation: 74 (20% split)
Training: 298 | Evaluation: 74
Labels: ner (4)
[2023-05-02 11:35:53,319] [INFO] Pipeline: ['tok2vec', 'ner']
[2023-05-02 11:35:53,321] [INFO] Created vocabulary
[2023-05-02 11:35:53,321] [INFO] Finished initializing nlp object
[2023-05-02 11:35:53,485] [INFO] Initialized pipeline components: ['tok2vec', 'ner']
✔ Initialized pipeline

============================= Training pipeline =============================
Components: ner
Merging training and evaluation data for 1 components
  - [ner] Training: 298 | Evaluation: 74 (20% split)
Training: 298 | Evaluation: 74
Labels: ner 

ℹ Using CPU

========================= Generating Prodigy config =========================
ℹ Auto-generating config with spaCy
✔ Generated training config

=========================== Initializing pipeline ===========================
[2023-05-02 14:56:44,408] [INFO] Set up nlp object from config
Components: ner
Merging training and evaluation data for 1 components
  - [ner] Training: 298 | Evaluation: 74 (20% split)
Training: 298 | Evaluation: 74
Labels: ner (4)
[2023-05-02 14:56:44,459] [INFO] Pipeline: ['tok2vec', 'ner']
[2023-05-02 14:56:44,463] [INFO] Created vocabulary
[2023-05-02 14:56:44,464] [INFO] Finished initializing nlp object
[2023-05-02 14:56:44,706] [INFO] Initialized pipeline components: ['tok2vec', 'ner']
✔ Initialized pipeline

============================= Training pipeline =============================
Components: ner
Merging training and evaluation data for 1 components
  - [ner] Training: 298 | Evaluation: 74 (20% split)
Training: 298 | Evaluation: 74
Labels: ner (4)
ℹ Pipeline: ['tok2vec', 'ner']
ℹ Initial learn rate: 0.001
E    #       LOSS TOK2VEC  LOSS NER  ENTS_F  ENTS_P  ENTS_R  SCORE 
---  ------  ------------  --------  ------  ------  ------  ------
  0       0          0.00     56.00    0.00    0.00    0.00    0.00
  7     200         70.53   1779.85   69.47   89.19   56.90    0.69
 17     400         17.79     54.66   74.00   88.10   63.79    0.74
 28     600          4.55      4.05   70.71   85.37   60.34    0.71
 42     800          4.28      2.71   74.51   86.36   65.52    0.75
 60    1000          0.09      0.04   73.08   82.61   65.52    0.73
 81    1200          4.32      1.62   72.16   89.74   60.34    0.72
106    1400         43.62     13.44   75.25   88.37   65.52    0.75
138    1600         66.88     27.63   70.97   94.29   56.90    0.71
175    1800         34.79      7.93   72.34   94.44   58.62    0.72
222    2000        184.67     43.45   70.21   91.67   56.90    0.70
276    2200         34.44     12.67   70.21   91.67   56.90    0.70
343    2400          5.49      0.61   71.58   91.89   58.62    0.72
410    2600         56.66     11.28   70.21   91.67   56.90    0.70
476    2800         51.57     10.86   66.67   88.57   53.45    0.67
543    3000         70.47      6.02   68.75   86.84   56.90    0.69

In [4]:
!python -m prodigy db-in news_train train.jsonl

✔ Created dataset 'news_train' in database SQLite
✔ Imported 273 annotations to 'news_train' (session
2023-05-02_11-40-37) in database SQLite
Found and keeping existing "answer" in 273 examples


In [5]:
!python -m prodigy db-in news_eval eval.jsonl

✔ Created dataset 'news_eval' in database SQLite
✔ Imported 100 annotations to 'news_eval' (session 2023-05-02_11-40-38)
in database SQLite
Found and keeping existing "answer" in 100 examples


In [6]:
!python -m prodigy train --ner news_train,eval:news_eval

ℹ Using CPU

========================= Generating Prodigy config =========================
ℹ Auto-generating config with spaCy
✔ Generated training config

=========================== Initializing pipeline ===========================
[2023-05-02 11:40:39,673] [INFO] Set up nlp object from config
Components: ner
Merging training and evaluation data for 1 components
  - [ner] Training: 273 | Evaluation: 99 (from datasets)
Training: 273 | Evaluation: 99
Labels: ner (4)
[2023-05-02 11:40:39,692] [INFO] Pipeline: ['tok2vec', 'ner']
[2023-05-02 11:40:39,694] [INFO] Created vocabulary
[2023-05-02 11:40:39,694] [INFO] Finished initializing nlp object
[2023-05-02 11:40:39,843] [INFO] Initialized pipeline components: ['tok2vec', 'ner']
✔ Initialized pipeline

============================= Training pipeline =============================
Components: ner
Merging training and evaluation data for 1 components
  - [ner] Training: 273 | Evaluation: 99 (from datasets)
Training: 273 | Evaluation: 99
Labe

ℹ Using CPU

========================= Generating Prodigy config =========================
ℹ Auto-generating config with spaCy
✔ Generated training config

=========================== Initializing pipeline ===========================
[2023-05-02 15:04:48,073] [INFO] Set up nlp object from config
Components: ner
Merging training and evaluation data for 1 components
  - [ner] Training: 273 | Evaluation: 99 (from datasets)
Training: 273 | Evaluation: 99
Labels: ner (4)
[2023-05-02 15:04:48,125] [INFO] Pipeline: ['tok2vec', 'ner']
[2023-05-02 15:04:48,129] [INFO] Created vocabulary
[2023-05-02 15:04:48,130] [INFO] Finished initializing nlp object
[2023-05-02 15:04:48,357] [INFO] Initialized pipeline components: ['tok2vec', 'ner']
✔ Initialized pipeline

============================= Training pipeline =============================
Components: ner
Merging training and evaluation data for 1 components
  - [ner] Training: 273 | Evaluation: 99 (from datasets)
Training: 273 | Evaluation: 99
Labels: ner (4)
ℹ Pipeline: ['tok2vec', 'ner']
ℹ Initial learn rate: 0.001
E    #       LOSS TOK2VEC  LOSS NER  ENTS_F  ENTS_P  ENTS_R  SCORE 
---  ------  ------------  --------  ------  ------  ------  ------
  0       0          0.00     56.11    0.00    0.00    0.00    0.00
  8     200         67.68   1706.38   35.05   58.62   25.00    0.35
 19     400         48.75     96.31   42.11   74.07   29.41    0.42
 31     600          4.30      7.75   43.14   64.71   32.35    0.43
 47     800          0.20      0.09   40.40   64.52   29.41    0.40
 66    1000         21.49      7.34   38.00   59.38   27.94    0.38
 89    1200          0.85      0.48   36.89   54.29   27.94    0.37
118    1400         25.89      9.57   33.33   45.00   26.47    0.33
152    1600        118.67     50.31   37.50   64.29   26.47    0.37
193    1800         45.59     14.22   38.00   59.38   27.94    0.38
243    2000         14.89      4.81   36.04   46.51   29.41    0.36
306    2200         19.05      3.07   46.02   57.78   38.24    0.46
373    2400         24.70      3.84   42.59   57.50   33.82    0.43
440    2600         69.18     10.95   40.82   66.67   29.41    0.41
506    2800        155.87     20.43   36.54   52.78   27.94    0.37
573    3000          6.48      1.67   39.18   65.52   27.94    0.39
640    3200        237.26     31.02   36.96   70.83   25.00    0.37
706    3400        198.38     29.66   48.54   71.43   36.76    0.49
773    3600          0.00      0.00   48.08   69.44   36.76    0.48
840    3800         38.16      4.01   44.90   73.33   32.35    0.45
906    4000          0.00      0.00   44.44   70.97   32.35    0.44
973    4200         14.14      1.37   40.35   50.00   33.82    0.40
1040    4400          1.39      0.14   42.42   67.74   30.88    0.42
1106    4600        480.32     58.45   44.68   80.77   30.88    0.45
1173    4800          6.54      0.70   43.96   86.96   29.41    0.44
1240    5000        106.19      8.15   44.44   90.91   29.41    0.44